query-planner-js/src/buildPlan.ts

```diff
@@ -800,7 +800,8 @@ class FetchGroup {
       serviceName: this.subgraphName,
       requires: inputNodes ? trimSelectionNodes(inputNodes.selections) : undefined,
       variableUsages: this.selection.usedVariables().map(v => v.name),
-      operation: stripIgnoredCharacters(print(operation)),
+      operation: stripIgnoredCharacters(print(operationToDocument(operation))),
+      operationKind: operation.rootKind as OperationTypeNode,
```

Comment:
I assumed as well, but I'm getting the following error:

```
query-planner-js/src/buildPlan.ts:804:7 - error TS2322: Type 'SchemaRootKind' is not assignable to type 'OperationTypeNode'.
  Type '"query"' is not assignable to type 'OperationTypeNode'.

804       operationKind: operation.rootKind,// as OperationTypeNode,
          ~~~~~~~~~~~~~

  query-planner-js/src/QueryPlan.ts:34:3
    34   operationKind: OperationTypeNode;
         ~~~~~~~~~~~~~
    The expected type comes from property 'operationKind' which is declared here on type 'FetchNode'
```

is it forbidden to cast between unions and enums?

Reply:
Oh, my bad. `OperationTypeNode` used to be just a string union type, but they changed it to an enum in graphql-js 16 ...

Then, I don't understand typescript enum all that well and I'm not 100% surprised that casting might work but enum values are supposed to be opaque, so it would probably be cleaner to do a genuine conversion rather than casting. Though I guess longer time I'll just remove `SchemaRootKind` in favour of using `OperationTypeNode` directly, but you may not want to bother with that here.